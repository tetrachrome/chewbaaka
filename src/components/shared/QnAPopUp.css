/**
 * QnAPopUp.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Aug 07, 2020
 */

div.QnAPopUpContainer {
  width: 180px;
  height: 60px;
  position: relative;
}

div.QnAPopUpMainTextBubbleContainer {
  width: 320px;
  height: 320px;
  position: relative;
  left: 200px;
  top: -420px;
  z-index: 11;
}

div.QnAPopUpTextSecondaryBubbleContainer {
  width: 45px;
  height: 45px;
  position: relative;
  left: 170px;
  top: -110px;
  z-index: 11;
}

div.QnAPopUpButtonGroupContainer {
  display: inline-block;
  z-index: 9;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 18px;
  font-weight: 600;
  color: #212529;
}

button.ui.orange.circular.icon.button.QuestionButton {
  margin-left: 10px;
}

button.ui.black.circular.icon.button.QuestionButton {
  margin-left: 10px;
}
