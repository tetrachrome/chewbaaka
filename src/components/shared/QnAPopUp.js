/**
 * QnAPopUp.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Jul 06, 2020
 */

import React from 'react';

import "semantic-ui-css/semantic.min.css";

import { Button } from 'semantic-ui-react'

import './QnAPopUp.css'

import TextBubble from './TextBubble'

class QnAPopUp extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      isOpen: false
    }
    this.handleToggle = this.handleToggle.bind(this);
  }

  handleToggle() {
    this.setState((prevState) => ({
      isOpen: prevState.isOpen ^ 1,
    }));
  }

  render() {
    return (
      <div className="QnAPopUpContainer">
        <div className="QnAPopUpMainTextBubbleContainer" style={{width: 256, height:256}}>
          <TextBubble hidden={!this.state.isOpen} diameter={256} title="Did you know?" content={this.props.content}/>
        </div>
        <div className="QnAPopUpTextSecondaryBubbleContainer" style={{width: 45, height:45}}>
          <TextBubble hidden={!this.state.isOpen} diameter={45} title="" content=""/>
        </div>
        <div className="QnAPopUpButtonGroupContainer">
          <Button className="QuestionButton" circular icon="question" color={this.state.isOpen ? "black" : "orange"} onClick={this.handleToggle} />
          {/* <span>Did you know?</span> */}
        </div>
      </div>
    );
  }
}

export default QnAPopUp;
