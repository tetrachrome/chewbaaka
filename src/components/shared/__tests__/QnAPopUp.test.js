/**
 * QnAPopUp.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Jul 06, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import QnAPopUp from '../QnAPopUp';

test('renders QnAPopUp component', () => {
  const content = "The cheetah is the fastest land animal!";
  render(<QnAPopUp content={content}/>);

  const contentElement = screen.getByText(content);
  expect(contentElement).toBeInTheDocument();
});
