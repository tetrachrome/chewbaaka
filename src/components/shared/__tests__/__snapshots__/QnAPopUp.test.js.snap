// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QnAPopUp component snapshot 1`] = `
<div
  className="QnAPopUpContainer"
>
  <div
    className="QnAPopUpMainTextBubbleContainer"
    style={
      Object {
        "height": 256,
        "width": 256,
      }
    }
  >
    <div
      className="TextBubbleOuterContainer"
      hidden={true}
      style={
        Object {
          "backgroundColor": "rgb(undefined)",
          "height": 256,
          "width": 256,
        }
      }
    >
      <div
        className="TextBubbleInnerContainer"
      >
        <div
          className="TextBubbleInnerCore"
        >
          <h1
            style={
              Object {
                "color": "rgb(undefined)",
                "fontSize": 20.48,
              }
            }
          >
            Did you know?
          </h1>
          <p
            style={
              Object {
                "color": "rgb(undefined)",
                "fontSize": 12.8,
              }
            }
          >
            The cheetah is the fastest land animal!
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="QnAPopUpTextSecondaryBubbleContainer"
    style={
      Object {
        "height": 45,
        "width": 45,
      }
    }
  >
    <div
      className="TextBubbleOuterContainer"
      hidden={true}
      style={
        Object {
          "backgroundColor": "rgb(undefined)",
          "height": 45,
          "width": 45,
        }
      }
    >
      <div
        className="TextBubbleInnerContainer"
      >
        <div
          className="TextBubbleInnerCore"
        >
          <h1
            style={
              Object {
                "color": "rgb(undefined)",
                "fontSize": 3.6,
              }
            }
          >
            
          </h1>
          <p
            style={
              Object {
                "color": "rgb(undefined)",
                "fontSize": 2.25,
              }
            }
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="QnAPopUpButtonGroupContainer"
  >
    <button
      className="ui orange circular icon button QuestionButton"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="question icon"
        onClick={[Function]}
      />
    </button>
  </div>
</div>
`;
